src/core/runtime.cpp

```diff
@@ -1500,22 +1503,19 @@ SHTypeInfo deriveTypeInfo(const SHVar &value, const SHInstanceData &data, bool *
     s.api->setGetIterator(s, &sit);
     SHVar v;
     while (s.api->setNext(s, &sit, &v)) {
-      auto derived = deriveTypeInfo(v, data, containsVariables);
+      auto derived = deriveTypeInfo(v, data, expInfo);
       shards::arrayPush(varType.setTypes, derived);
     }
   } break;
   case SHType::ContextVar: {
-    if (containsVariables) {
-      // containsVariables is used by Const shard mostly
-      *containsVariables = true;
+    if (expInfo) {
       const auto varName = value.payload.stringValue;
       for (auto info : data.shared) {
         if (strcmp(info.name, varName) == 0) {
+          expInfo->push_back(SHExposedTypeInfo{.name = info.name, .exposedType = info.exposedType});
```

Comment:
wrap around  `if(expInfo)` here, so the value is still returned when it's null

Reply:
Good point but I'm afraid sometimes we actually want to return as it is ContextVar as result/type... Wonder if we have coverage for that.
For some reasons the previous version was doing the same (with a bool switch)

Reply:
Hmm yeah I see it was like that already. okay fine.